DB設計

y1mutsu opened the pull request:
# WHAT
Chat-spaceのDB設計

y1mutsu committed:
[WIP]DB設計

# WHAT
 Chat-spaceのDB設計

kuroko918 commented on README.md:
```diff
@@ -1,24 +1,48 @@
 
-* Deployment instructions
+### Association
+- has_many :texts
```
アソシエーションはそもそもですが、モデル間の関係を示すものなので、こちらは必要ないかと思います。

kuroko918 commented on README.md:
```diff
@@ -1,24 +1,48 @@
-* Database initialization
 
-* How to run the test suite
+## messageテーブル
```
groupとのアソシエーションが抜けているので消しておきましょう！

y1mutsu replied:
messageテーブルを消すとは、何故かおしえていただけますでしょうか？

kuroko918 commented on README.md:
```diff
@@ -1,24 +1,48 @@
+
+### Association
+- has_many :groups
+- belongs_to :user
```
belongsではないですね！
user has_many groupsであり、group has_many usersです！

kuroko918 commented on README.md:
```diff
@@ -1,24 +1,48 @@
+|Column|Type|Options|
+|------|----|-------|
+|user|reference|null: false, foreign_key: true|
+|e-mail|reference|null: false, foreign_key: true|
```
reference型は外部キーに使用する型です。
string型にしましょう！

y1mutsu replied:
修正しました

yusukeyamane commented:
最新版のコードがプッシュされていないようです

y1mutsu committed:
[WID]chat-space　DB設計

# What DB設　修正
# How
4箇所訂正しました。

中間テーブル　throughを使用して記載

※messageテーブル消してくださいとのことでしたが、必要かと思い修正して残しました。消す理由をお教えいただけますでしょうか？

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,47 @@
+|password|integer|null: false|
 
-* Database creation
+### Association
```
messagesとのアソシエーションが欠けてますね

y1mutsu replied:
修正しました

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,47 @@
-* Database initialization
 
-* How to run the test suite
+## messagesテーブル
```
messageがuserの情報しか持ってないので、groupsともアソシエーションを組む必要があります

y1mutsu replied:
修正しました

TOMARUMARU replied:
こちら修正されていません

y1mutsu committed:
[WID]chat-space DB設計(3)

# what
Chat-space DB設計

#how
2箇所訂正しました

TOMARUMARU commented:
pushしてくだい

miyagit commented on README.md:
```diff
@@ -1,24 +1,48 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|user|reference|null: false, foreign_key: true|
```
こちらgroupsてーぶるにもuser_idとgroup_idが必要なのでしょうか？

y1mutsu replied:
修正しました

miyagit reviewed (commented):
すでに記述されているコメントが修正されていないのでそちら修正してください。

y1mutsu committed:
[WID]chat-spaceDB設計

# What
Caht-spaceDB設計　修正しました

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,49 @@
-* ...
+### Association
+- belongs_to :user
+- has_many :members
```
https://github.com/y1mutsu/chat-space4/pull/1/files#r258383066
治ってないです

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,49 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false|
```
nameでいいです

y1mutsu replied:
修正しました

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,49 @@
-* Configuration
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false|
```
index追加

y1mutsu replied:
emailに追加しました

y1mutsu committed:
[WID]cahtspace DB設計

# what
DB設計　修正しました

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false|
+|e-mail|string|null: false,add_index :email|
```
indexはemailではなく、nameですね

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|------|----|-------|
+|name|string|null: false|
+
+### Association
```
messagesとのアソシエーションがありません

y1mutsu committed:
[WDI]chat-spaceのDB設計

#What
Chat-space DB設計　修正しました

yusukeyamane commented:
LGTMです
